pkg/sqlmanager/build_single_select_field.go

```diff
@@ -0,0 +1,115 @@
+package sqlmanager
+
+import (
+	"fmt"
+	"github.com/nrc-no/core/pkg/api/types"
+	"github.com/nrc-no/core/pkg/sql/schema"
+)
+
+// singleSelectFieldActions returns the SQL actions necessary to build a SingleSelect field
+// When a user creates a form with a Single Select Field, there will be an SQL Table created
+// to simply hold the different options for that field (ID and Name).
+//
+// We also add sqlActions required to populate this table with the options from the field definition.
+//
+// Also, we add a column to the Form SQL Table to hold the currently selected option.
+// This allows us to add a Foreign Key to that field's options table for referential integrity.
+//
+// The options table looks like so
+//
+// Table Name: <FIELD_ID>_options
+// +=======+==========+
+// | id    | name     |
+// +=======+==========+
+// | uid1  | Option 1 |
+// +-------+----------+
+// | uid2  | Option 2 |
+// +-------+----------+
+//
+func singleSelectFieldActions(formInterface types.FormInterface, fieldDefinition *types.FieldDefinition) (sqlActions, error) {
+	result := sqlActions{}
+	result = append(result, buildSelectOptionsTable(formInterface, fieldDefinition)...)
+
+	sqlField := getStandardSQLColumnForField(fieldDefinition)
+	sqlField.DataType = schema.SQLDataType{
+		VarChar: &schema.SQLDataTypeVarChar{
+			Length: 36,
```

Comment:
could this 36 be moved to a variable, since it seems to be used in a lot of the field builders?

Reply:
Yes, will do that